ruby-on-rails/app/controllers/api/v1/checkout_controller.rb

```diff
@@ -75,16 +80,19 @@ def index
     #default message, rewritten if exception
     logged = "Thanks for your order. Total cost is $" + total 
 
+    Sentry.logger.trace("Starting inventory database connection")
     span_inventory_call = transaction.start_child(op: "custom.inventory_db_call")
     products_in_inventory = Inventory.all()
     span_inventory_call.finish
+    Sentry.logger.trace("Completed inventory database query, found %{count} items", count: products_in_inventory.length)
 
     span_logic = transaction.start_child(op: "custom.inventory_vs_cart_logic")
 
     products_in_inventory.each_with_index { |inv_objs, i|
       if !enough_inventory?(cart_contents)
         begin
-          raise Exception.new "Not enough inventory for product"
+          Sentry.logger.error("Failed to process payment. Insufficient inventory for product: %{product_id}", product_id: inv_objs["productid"])
+          raise Exception.new "Not enough inventory for product: #{inv_objs["productid"]}"
```

Comment:
if you're raising one here, I'd remove the explicit `Sentry.capture_exception` below on line 97.